From bae898598a344501436d2e3cfcb2cc48e9d0b77b Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Fri, 31 Oct 2014 21:34:53 -0400
Subject: [tools] testing/selftests: Add infrastructure for powerpc selftests

Message-id: <1414791294-15553-2-git-send-email-gduarte@redhat.com>
Patchwork-id: 98661
O-Subject: [RHEL7.1 PATCH BZ 1134511 1/2] selftests: Add infrastructure for powerpc selftests
Bugzilla: 1134511
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: 1134511
Upstream Status: 0e56dacdda49940ff6e24e504f11468a27922416

Required for backporting 96d016108640bc2b7f

commit 0e56dacdda49940ff6e24e504f11468a27922416
Author: Michael Ellerman <michael@ellerman.id.au>
Date:   Tue Aug 6 17:42:35 2013 +1000

    selftests: Add infrastructure for powerpc selftests

    This commit adds a powerpc subdirectory to tools/testing/selftests,
    for tests that are powerpc specific.

    On other architectures nothing is built. The makefile supports cross
    compilation if the user sets ARCH and CROSS_COMPILE.

    Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/testing/selftests/Makefile b/tools/testing/selftests/Makefile
index 0a63658..c7fd8ac 100644
--- a/tools/testing/selftests/Makefile
+++ b/tools/testing/selftests/Makefile
@@ -7,6 +7,7 @@ TARGETS += mqueue
 TARGETS += net
 TARGETS += ptrace
 TARGETS += vm
+TARGETS += powerpc
 
 all:
  for TARGET in $(TARGETS); do \
diff --git a/tools/testing/selftests/powerpc/Makefile b/tools/testing/selftests/powerpc/Makefile
new file mode 100644
index 0000000..b315740
--- /dev/null
+++ b/tools/testing/selftests/powerpc/Makefile
@@ -0,0 +1,39 @@
+# Makefile for powerpc selftests
+
+# ARCH can be overridden by the user for cross compiling
+ARCH ?= $(shell uname -m)
+ARCH := $(shell echo $(ARCH) | sed -e s/ppc.*/powerpc/)
+
+ifeq ($(ARCH),powerpc)
+
+GIT_VERSION = $(shell git describe --always --long --dirty || echo "unknown")
+
+CC := $(CROSS_COMPILE)$(CC)
+CFLAGS := -Wall -O2 -flto -Wall -Werror -DGIT_VERSION='"$(GIT_VERSION)"' -I$(CURDIR) $(CFLAGS)
+
+export CC CFLAGS
+
+TARGETS =
+
+endif
+
+all:
+ @for TARGET in $(TARGETS); do \
+  $(MAKE) -C $$TARGET all; \
+ done;
+
+run_tests: all
+ @for TARGET in $(TARGETS); do \
+  $(MAKE) -C $$TARGET run_tests; \
+ done;
+
+clean:
+ @for TARGET in $(TARGETS); do \
+  $(MAKE) -C $$TARGET clean; \
+ done;
+ rm -f tags
+
+tags:
+ find . -name '*.c' -o -name '*.h' | xargs ctags
+
+.PHONY: all run_tests clean tags
-- 
1.7.1